cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/ApacheKafkaTopicService.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    ListTopicsOptions listTopicsOptions = new ListTopicsOptions();
+    listTopicsOptions = listTopicsOptions.listInternal(false);
+
+    ListTopicsResult topicsResult = client.listTopics(listTopicsOptions);
+
+    try {
+      DescribeTopicsResult s = client.describeTopics(new ArrayList<>(topicsResult.names().get()));
+      Map<String, TopicDescription> topicDesc =
+          s.all().get(TIME_OUT_SECS_FOR_TOPICS, TimeUnit.SECONDS);
+      Set<String> keySet = topicDesc.keySet();
+      keySet.remove("_schemas");
+      List<String> lstK = new ArrayList<>(keySet);
+      HashMap<String, String> hashMap;
+      for (String topicName : lstK) {
+        hashMap = new HashMap<>();
+        hashMap.put("topicName", topicName);
+        hashMap.put(
+            "replicationFactor",
+            "" + topicDesc.get(topicName).partitions().get(0).replicas().size());
+        hashMap.put("partitions", "" + topicDesc.get(topicName).partitions().size());
+        if (!topicName.startsWith("_confluent")) {
+          topics.add(hashMap);
+        }
+      }
+
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      log.error(e.getMessage());
+    }
+    return topics;
+  }
+
+  public synchronized ApiResponse createTopic(ClusterTopicRequest clusterTopicRequest)
+      throws Exception {
+    log.info("createTopic {}", clusterTopicRequest);
+    AdminClient client;
+    try {
+      client =
+          clusterApiUtils.getAdminClient(
+              clusterTopicRequest.getEnv(),
+              clusterTopicRequest.getProtocol(),
+              clusterTopicRequest.getClusterName());
+      if (client == null) {
+        throw new Exception("Cannot connect to cluster.");
+      }
+
+      NewTopic topic =
+          new NewTopic(
+              clusterTopicRequest.getTopicName(),
+              clusterTopicRequest.getPartitions(),
+              clusterTopicRequest.getReplicationFactor());
+
+      CreateTopicsResult result = client.createTopics(Collections.singletonList(topic));
+      result
+          .values()
+          .get(clusterTopicRequest.getTopicName())
+          .get(TIME_OUT_SECS_FOR_TOPICS, TimeUnit.SECONDS);
+    } catch (KafkaException e) {
+      String errorMessage = "Invalid properties: ";
+      log.error(errorMessage, e);
+      throw e;
+    } catch (NumberFormatException e) {
+      String errorMessage = "Invalid replica assignment string";
+      log.error(errorMessage, e);
+      throw e;
+    } catch (ExecutionException | InterruptedException e) {
+      String errorMessage;
+      if (e instanceof ExecutionException) {
+        errorMessage = e.getCause().getMessage();
+      } else {
+        Thread.currentThread().interrupt();
+        errorMessage = e.getMessage();
+      }
+      log.error("Unable to create topic {}, {}", clusterTopicRequest.getTopicName(), errorMessage);
+      throw e;
+    } catch (Exception e) {
+      log.error(e.getMessage());
```

Comment:
it's better to log the whole trace 
```suggestion
      log.error("Exception: ", e);
```